DNN Platform/DotNetNuke.Web.Mvc/Extensions/StartupExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using DotNetNuke.DependencyInjection.Extensions;
+using DotNetNuke.Web.Mvc.Framework.Controllers;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DotNetNuke.Web.Mvc.Extensions
+{
+    public static class StartupExtensions
+    {
+        public static void AddWebApiControllers(this IServiceCollection services)
+        {
+            var controllerTypes = AppDomain.CurrentDomain.GetAssemblies()
+                .SelectMany(TypeExtensions.SafeGetTypes)
+                .Where(x => typeof(IDnnController).IsAssignableFrom(x)
+                    && x.IsClass
+                    && !x.IsAbstract
+                );
+            foreach (var controller in controllerTypes)
+            {
+                services.AddScoped(controller);
```

Comment:
Controllers were explicitly removed from the Container because of memory leak problems. In earlier versions of Dependency Injection the controllers that got registered in the container would never be collected by the Garbage Collector. This means if you have 100,000 requests to a specific controller, you would have 100,000 instances of the controller in memory.

I am hesitant to add controllers back in unless there is a very good reason for it

Reply:
see https://github.com/dnnsoftware/Dnn.Platform/pull/3520#discussion_r370655221 for the reasoning behind the memory leak.

Reply:
perfect, thanks!

Reply:
I would like to re-do the same testing that was done to validate the fix that was put in place in 9.4.x to ensure that we don't have a leak, but this looks to be the better way regardless

Reply:
Please note that if the controllers are not registered in the container it will not be able to resolve them. Which means that in 9.4.4 (apart from the provider being always null) the controllers were always resolved from the asp net DefaultControllerFactory `_resolver` that was passed in, which only supports parameterless constructors, so no DI for them.